### PR TITLE
Adsk contrib - Fix issues when building OCIO with MS Visual Studio 2022

### PR DESCRIPTION
**Pull request by cedrik-fuoco-adsk**
All issues were discovered when trying to configure and build OCIO on Windows using Microsoft Visual Studio 2022.

Pybind11 has an issue with Microsoft Visual Studio 2022, FindGlew from CMake has a bug around setting GLEW_INCLUDE_DIRS correctly and OpenImageIO as some warnings that get flagged as error. Since those warning are in the OCIO apps, we decided that I would be ok to ignore them temporarily.

Content overview:

Updating Pybind11 to the latest version as Pybind11 <2.9.0 has issues with Microsoft Visual Studio 2022.
Add fix to make sure that GLEW_INCLUDE_DIRS is set correctly.
Ignoring warnings (C4267 and C4244) temporarily in ociolutimage, ocioconvert and oiiohelpers

Reference issue #1634.

Signed-off-by: Cedrik Fuoco <cedrik.fuoco@autodesk.com>

### CODE REVIEW

File: share/cmake/utils/CheckSupportGL.cmake [hunk 1]
```diff
@@ -35,6 +36,35 @@ if((OCIO_BUILD_TESTS AND OCIO_BUILD_GPU_TESTS) OR OCIO_BUILD_APPS)
                 set(GLEW_LIBRARIES GLEW::GLEW)
             endif()
         endif()
+
+        # Cmake has a bug with FindGLEW. It fails to defined GLEW_INCLUDE_DIRS in some situation.
```
Comment (remia) [inline share/cmake/utils/CheckSupportGL.cmake:40]: Question: Is the comment on line 33-34 still valid or should it be removed? It seems the CMake bug is still existing and not fixed as suggested on that comment.

File: share/cmake/utils/CheckSupportGL.cmake [hunk 2]
```diff
@@ -35,6 +36,35 @@ if((OCIO_BUILD_TESTS AND OCIO_BUILD_GPU_TESTS) OR OCIO_BUILD_APPS)
...
+            if("${_GLEW_DEFS}" MATCHES "GLEW_STATIC")
+                get_target_property(GLEW_LIBRARY_DEBUG GLEW::GLEW IMPORTED_LOCATION_DEBUG)
+                get_target_property(GLEW_LIBRARY_RELEASE GLEW::GLEW IMPORTED_LOCATION_RELEASE)
+            else()
+                get_target_property(GLEW_LIBRARY_DEBUG GLEW::GLEW IMPORTED_IMPLIB_DEBUG)
+                get_target_property(GLEW_LIBRARY_RELEASE GLEW::GLEW IMPORTED_IMPLIB_RELEASE)
+            endif()
+            get_target_property(_GLEW_LINK_INTERFACE GLEW::GLEW IMPORTED_LINK_INTERFACE_LIBRARIES_RELEASE) # same for debug and release
+            list(APPEND GLEW_LIBRARIES ${_GLEW_LINK_INTERFACE})
+            list(APPEND GLEW_LIBRARY ${_GLEW_LINK_INTERFACE})
```
Comment (remia) [inline share/cmake/utils/CheckSupportGL.cmake:58]: This looks like undefined in Linux which breaks CI, should this line be removed as we don't seem to use `GLEW_LIBRARY` elsewhere? Is this variable only created on Windows? CMake [documentation](https://cmake.org/cmake/help/latest/module/FindGLEW.html) doesn't mention it.

**Comment by remia** [comment]
Thanks @cedrik-fuoco-adsk, when the Linux issue has been resolved, I would suggest we merge this one quickly as the CI seem to be broken on other PR for Windows which is fixed here. Probably some update to the Windows runners (although they are not using VS 2022).

File: share/cmake/utils/CheckSupportGL.cmake (hunk 2 above)
Comment (cedrik-fuoco-adsk) [reply share/cmake/utils/CheckSupportGL.cmake:58]: You're right. I don't see it used anywhere. I've seen some mentioned of GLEW_LIBRARY, but they are all from before 2019/2018 with an older version of cmake.

I'll remove it as it is not needed on Windows either.

File: share/cmake/utils/CheckSupportGL.cmake (hunk 1 above)
Comment (cedrik-fuoco-adsk) [reply share/cmake/utils/CheckSupportGL.cmake:40]: I'll remove it. It is not valid anymore as that issue was closed by accident and re-opened.

**Review by doug-walker — APPROVED** [review]
Note:  This is a build-break fix and therefore the two-week rule does not apply.

**Pull request merged**